django/db/models/expressions.py

```diff
@@ -915,9 +916,13 @@ def get_source_expressions(self):
         return [self.expression]
 
     def get_group_by_cols(self, alias=None):
-        expression = self.expression.copy()
-        expression.output_field = self.output_field
-        return expression.get_group_by_cols(alias=alias)
+        # The entire expression must be included in the GROUP BY clause if it
+        # is an SQL WHERE clause.
+        if not isinstance(self.expression, Node):
```

Comment:
I don't have a strong opinion here as that's probably the only type of non-`Expression` wrapper supported in `ExpressionWrapper` but the previous `if isinstance(self.expression, Expression)` felt more correct.

Reply:
I have mixed feelings because in other places we check the `resolve_expression` attribute, which we cannot use here 🤔. Neither solution is perfect 😐

Reply:
yeah I think that the crux of the issue is that `Q` and what it resolves to are not expression here but that would be an invasive change to backport. Feels kind of strange to wrap a non-`Expression` in an abstraction called `ExpressionWrapper`.

Reply:
Changed.